src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -108,9 +114,9 @@ fn limit_filename_len(filename: String) -> String {
     }
 }
 
-impl CoverageCalculator {
-    fn new() -> CoverageCalculator {
-        CoverageCalculator { items: Default::default() }
+impl<'a, 'b> CoverageCalculator<'a, 'b> {
+    fn new(ctx: &'a DocContext<'b>) -> CoverageCalculator<'a, 'b> {
```

Comment:
```suggestion
    fn new(ctx: &'a DocContext<'b>) -> Self {
```

Reply:
I'd rather keep it the current way if you don't mind. When dealing with lifetimes, I prefer when it's explicit.

Reply:
Hmm, not my preference but I don't feel strongly about it.